piet/src/text.rs

```diff
@@ -94,6 +132,34 @@ pub trait TextLayout {
     fn hit_test_text_position(&self, text_position: usize) -> Option<HitTestTextPosition>;
 }
 
+/// Metadata about each line in a text layout.
+#[derive(Debug, Default)]
+pub struct LineMetric {
+    /// Line length (in code units), but offset from the beginning of the text. So it's the length
+    /// of this line summed with the lengths of all the lines before it.
+    ///
+    /// Does not include trailing whitespace.
+    pub line_length_offset: usize,
```

Comment:
I think the name here is confusing, maybe something more like `line_end_offset`.